File: web/src/graphql/mutations/ticket.ts
import { gql } from "@apollo/client";

export const CREATE_TICKET = gql`
  mutation CreateTicket(
    $startComment: String
    $date: DateTime!
    $status: Float!
    $shopId: ID!
    $posId: ID!
    $responsibleUserId: ID!
    $startMoney: Float!
  ) {
    createTicket(
      startComment: $startComment
      date: $date
      posId: $posId
      status: $status
      shopId: $shopId
      responsibleUserId: $responsibleUserId
      startMoney: $startMoney
    ) {
      id
    }
  }
`;

export const CREATE_TICKET_PRODUCTS = gql`
  mutation CreateTicketProducts(
    $productIds: [ID!]!
    $ticketId: ID!
    $startAmounts: [Float!]!
  ) {
    createTicketProducts(
      productIds: $productIds
      ticketId: $ticketId
      startAmounts: $startAmounts
    )
  }
`;

File: playground_app/lib/graphql/mutations/ticket.dart
String updateTicket = """
mutation updateTicket(
  \$id: ID!
  \$responsibleUserId: ID
  \$status: Float
  \$startMoney: Float
  \$endMoney: Float
  \$startComment: String
  \$midComment: String
  \$endComment: String
  \$posId: ID
) {
  updateTicket(
    id: \$id
    responsibleUserId: \$responsibleUserId
    status: \$status
    startMoney: \$startMoney
    endMoney: \$endMoney
    startComment: \$startComment
    midComment: \$midComment
    endComment: \$endComment
    posId: \$posId
  ) {
    id
  }
}
""";

String updateTicketProductsOfTicket = """
mutation updateTicketProductsOfTicket(
  \$productIds: [ID!]!
  \$ticketId: ID!
  \$endAmounts: [Float!]!
  
) {
  updateTicketProductsOfTicket(
    productIds: \$productIds
    ticketId: \$ticketId
    endAmounts: \$endAmounts
  ) 
}
""";
